components/color_input.tsx

```diff
@@ -2,37 +2,30 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import ReactDom from 'react-dom';
-import PropTypes from 'prop-types';
-import {ChromePicker} from 'react-color';
+import {ChromePicker, ColorResult} from 'react-color';
 
-class ColorInput extends React.Component {
-    static propTypes = {
-
-        /*
-        * The id of setting that we will change
-        */
-        id: PropTypes.string.isRequired,
+type Props = {
+    id: string;
+    color: string;
+    onChange?: (hex: string) => void;
+}
 
-        /*
-         * Selected color
-         */
-        color: PropTypes.string.isRequired,
+type State = {
+    isOpened: boolean;
+}
 
-        /*
-         * Function called when color changed. Takes hex format of color Ex: #ffeec0
-         */
-        onChange: PropTypes.func,
-    };
+class ColorInput extends React.Component<Props, State> {
```

Comment:
```suggestion
class ColorInput extends React.PureComponent<Props, State> {
```
Unless we rely on non-Pure for some reason

Reply:
Good point, there is not reason to move that into a PureComponent :)